application/src/main/java/org/thingsboard/server/service/security/model/token/JwtTokenFactory.java

```diff
@@ -161,6 +163,7 @@ public SecurityUser parseRefreshToken(RawAccessJwtToken rawAccessToken) {
         UserPrincipal principal = new UserPrincipal(isPublic ? UserPrincipal.Type.PUBLIC_ID : UserPrincipal.Type.USER_NAME, subject);
         SecurityUser securityUser = new SecurityUser(new UserId(UUID.fromString(claims.get(USER_ID, String.class))));
         securityUser.setUserPrincipal(principal);
+        securityUser.setSessionId(claims.get(SESSION_ID, String.class));
```

Comment:
Please check whether it is present in the token, because after the upgrade to 3.4.2 users will still be using old tokens with no sessionId in them